### DISCUSSION TITLE
Not able to render jupyter lab/notebooks in browser although the kernel is starting alright.

### DISCUSSION THREAD

**Discussion by binarycache** [gitpod-io/openvscode-server#319] [General] [answered]
- VS Code Version: 1.64.2 
- OS Version: 18.04.6 LTS

Steps to Reproduce:

1. Running the docker image from inside slurm accessed via ssh.
2. docker run -it --init -p 3000:3000 -v "$(pwd):/home/workspace:cached" gitpod/openvscode-server
3. Afte installing python kernel is getting selected.
4. Can't render ipynb files (Tried on different versions of python, even with conda environment)

Jupyter Log Output:
Info 15:40:06: Find preferred kernel for untitled:Untitled-1.ipynb?jupyter-notebook with metadata {"orig_nbformat":4,"language_info":{"name":"python"}} & preferred interpreter /usr/bin/python3
Info 15:40:06: findKernel found Python 3.6.9 64-bit
Info 15:40:06: PreferredConnection: .jvsc74a57bd031f2aee4e71d21fbe5cf8b01ff0e069b9275f58929596ceb00d14d90e3e16cd6./usr/bin/python3./usr/bin/python3.-m#ipykernel_launcher found for NotebookDocument: untitled:Untitled-1.ipynb?jupyter-notebook
Info 15:40:06: TargetController found ID: .jvsc74a57bd031f2aee4e71d21fbe5cf8b01ff0e069b9275f58929596ceb00d14d90e3e16cd6./usr/bin/python3./usr/bin/python3.-m#ipykernel_launcher for document untitled:Untitled-1.ipynb?jupyter-notebook
Info 15:40:06: Setting controller affinity for untitled:Untitled-1.ipynb?jupyter-notebook .jvsc74a57bd031f2aee4e71d21fbe5cf8b01ff0e069b9275f58929596ceb00d14d90e3e16cd6./usr/bin/python3./usr/bin/python3.-m#ipykernel_launcher
Info 15:40:06: Creating controller for jupyter-notebook with interpreter /usr/bin/python3

Regards,
Anmol

**Comment by nooobcoder** [comment]
That's surely a websocket and HTTPs issue, you need to setup a NGINX/Apache2 reverse proxy with SSL to load webviews in your browser. #96 This comment is very hot to solve your solution, although I could not find it helpful, as this does not explain from basics. And I wanted to setup a reverse proxy using docker.
If webviews are the problem, then this is the only solution for now. The jupyter extension in vscode uses webviews to render a notebook. @binarycache

**Comment by nooobcoder** [comment]
You can also hop into the Chrome dev tools and check for webviews not being loaded.
Thanks

**Comment by augustoicaro** [comment, marked as answer]
I had the same problem. The simple direct solution is to make a `ssh` with tunnel `-L` to slrum node machine to the port that openvscode-server was started (usually 3000), something like: `ssh -L 3000:slrum_node:3000 user@slrum_node`. And then, in your browser access as `localhost` (`localhost:3000`).

When running applications in localhost no HTTPS is required to activate the web workers and the notebook will render as expected :)

This was a problem for me when I was trying to find an IDE solution to my environment. After dropping openvscode-server solution because I was not able to render the notebooks, some hours later and other failed strategies I found somewhere recommended to use [caddy server](https://caddyserver.com). Then searching more I found that you can run web workers in localhost without HTTPS.

**Comment by nooobcoder** [comment]
@augustoicaro Thanks it worked, i had no idea that SSH tunnels, fixes HTTPs issues.
This is not a permafix, but still would get my job easily done, thank you so much @augustoicaro

**Comment by nooobcoder** [comment]
Yes @augustoicaro all works now, I have hosted it as a docker container and accessing remotely.
Thanks for your solution @augustoicaro, worked out like a charm!
Any mods reading this PIN his solution too or upvote as much possible.

**Comment by augustoicaro** [comment]
@nooobcoder I said the same thing that isn't a permafix, but create a https server was never prioritized since that tunnel solution is working well...

**Comment by binarycache** [comment]
@augustoicaro Worked liked a charm! @nooobcoder 
Note: You might need to do ssh -L 3000:ip_address_ssh_server:3000 user@ssh_server instead of using the name of the server for port tunneling. 
P.S. I still got an error of webview permissions which was basically a problem with "brave's" shields, try turning off the shields, and allowing cookies and javacript. I switched to chrome though for quick testing.
P.P.S I am still getting an error saying couldn't start python client tools which might be an issue with language server but it's working fine.

Thanks again!

**Comment by nooobcoder** [comment]
@binarycache that was very predictable. You can either use the **FQDN or the IP address** of the server whose port you want to forwarded to your local host.
You can read more about SSH port forwarding on: [linuxize](https://linuxize.com/post/how-to-setup-ssh-tunneling/)
Good day. 🌻

**Comment by nooobcoder** [comment]
@binarycache 

`P.P.S I am still getting an error saying couldn't start python client tools which might be an issue with language server but it's working fine.`

Please provide further logs to track this down

**Comment by binarycache** [comment]
@nooobcoder I am not sure how to get the logs from it but the issue is and its solution is mentioned here https://github.com/Microsoft/vscode-python/issues/2877

**Comment by jeanp413** [comment]
Thanks @nooobcoder and @augustoicaro for jumping in and helping @binarycache :rocket: 
Though in my testing it seems notebooks are not working at least when using `https` behind a reverse proxy :thinking: in latest stable version `1.64.2`, notebooks work fine in `1.63.2`, will investigate more.
Logs from chrome console:
```
 ERR Notebook not found: vscode-remote://3000-databooth-publichellogit-pdxbgapzihl.ws-us32.gitpod.io/workspace/public-hello-gitpod/test-python.ipynb: createCellExecution@https://3000-databooth-publichellogit-pdxbgapzihl.ws-us32.gitpod.io/static/out/vs/workbench/workbench.web.main.js:1198:98987
$createExecution@https://3000-databooth-publichellogit-pdxbgapzihl.ws-us32.gitpod.io/static/out/vs/workbench/workbench.web.main.js:1534:27878
_doInvokeHandler@https://3000-databooth-publichellogit-pdxbgapzihl.ws-us32.gitpod.io/static/out/vs/workbench/workbench.web.main.js:1541:13103
_invokeHandler@https://3000-databooth-publichellogit-pdxbgapzihl.ws-us32.gitpod.io/static/out/vs/workbench/workbench.web.main.js:1541:12787
_receiveRequest@https://3000-databooth-publichellogit-pdxbgapzihl.ws-us32.gitpod.io/static/out/vs/workbench/workbench.web.main.js:1541:11449
_receiveOneMessage@https://3000-databooth-publichellogit-pdxbgapzihl.ws-us32.gitpod.io/static/out/vs/workbench/workbench.web.main.js:1541:10127
u/<@https://3000-databooth-publichellogit-pdxbgapzihl.ws-us32.gitpod.io/static/out/vs/workbench/workbench.web.main.js:1541:8236
fire@https://3000-databooth-publichellogit-pdxbgapzihl.ws-us32.gitpod.io/static/out/vs/workbench/workbench.web.main.js:67:1712
fire@https://3000-databooth-publichellogit-pdxbgapzihl.ws-us32.gitpod.io/static/out/vs/workbench/workbench.web.main.js:83:18808
_receiveMessage@https://3000-databooth-publichellogit-pdxbgapzihl.ws-us32.gitpod.io/static/out/vs/workbench/workbench.web.main.js:83:23389
f/<@https://3000-databooth-publichellogit-pdxbgapzihl.ws-us32.gitpod.io/static/out/vs/workbench/workbench.web.main.js:83:20923
fire@https://3000-databooth-publichellogit-pdxbgapzihl.ws-us32.gitpod.io/static/out/vs/workbench/workbench.web.main.js:67:1712
acceptChunk@https://3000-databooth-publichellogit-pdxbgapzihl.ws-us32.gitpod.io/static/out/vs/workbench/workbench.web.main.js:83:15639
o/<@https://3000-databooth-publichellogit-pdxbgapzihl.ws-us32.gitpod.io/static/out/vs/workbench/workbench.web.main.js:83:14769
onData/<@https://3000-databooth-publichellogit-pdxbgapzihl.ws-us32.gitpod.io/static/out/vs/workbench/workbench.web.main.js:585:41696
fire@https://3000-databooth-publichellogit-pdxbgapzihl.ws-us32.gitpod.io/static/out/vs/workbench/workbench.web.main.js:67:1712
E/this._fileReader.onload@https://3000-databooth-publichellogit-pdxbgapzihl.ws-us32.gitpod.io/static/out/vs/workbench/workbench.web.main.js:585:39626
```

**Comment by nooobcoder** [comment]
@jeanp413 can i have a easy guide with Docker compose to have a reverse proxy with websockets for openvscode server. I want a least brainf**k method of rev proxy using docker.

Also that would work with self signed certs without a domain name.
I just want https without much jargon.

**Comment by jeanp413** [comment]
@nooobcoder cannot promise anything but I agree it would be really helpful to have a docker compose config in the docs

**Comment by augustoicaro** [comment]
> @augustoicaro Worked liked a charm! @nooobcoder Note: You might need to do ssh -L 3000:ip_address_ssh_server:3000 user@ssh_server instead of using the name of the server for port tunneling. P.S. I still got an error of webview permissions which was basically a problem with "brave's" shields, try turning off the shields, and allowing cookies and javacript. I switched to chrome though for quick testing. P.P.S I am still getting an error saying couldn't start python client tools which might be an issue with language server but it's working fine.
> 
> Thanks again!

@binarycache About python language service, it will not work because it is not installed with Microsoft python extension, due to proprietary restrictions as remote extensions. So I use `Jedi` as a language service in my projects.

>Though in my testing it seems notebooks are not working at least when using https behind a reverse proxy thinking in latest stable version 1.64.2, notebooks work fine in 1.63.2, will investigate more.

@jeanp413, I'm using version 1.64.2 and it is working well with the ssh tunnel and accessing via localhost. The only problem I got in migration to 1.63.2 to 1.64.2 was the new way to start the server and now I need to use `--host machine_name` to access the server through the network.

**Comment by ayan2k26** [comment]
I am using 1.66.1 and I am unable to view notebooks. The command I used was `./openvscode-server-v1.66.1-linux-x64/bin/openvscode-server --host 192.168.3.63 --port 8585` and then `ssh -L 8585:192.168.3.63:8585 saifk@192.168.3.63`
I tried this with and without the `host` flag.
Without the host flag, the browsers (Brave, Firefox) couldn't connect to it with this error in the terminal with the `ssh...` command - `channel 3: open failed: connect failed: Connection refused`
This is what's coming in the output tab of log(window) when I am using the host flag and trying to open an ipynb file - 
```
[2022-04-10 11:00:16.414] [window] [info] Ignoring configured affinity for 'pub.name' because the extension is unknown or cannot execute.
[2022-04-10 11:00:17.288] [window] [error] Can't compute sha-256: Error: Can't compute sha-256
    at t (http://192.168.3.63:8585/static/out/vs/workbench/workbench.web.main.js:797:37794)
    at new F (http://192.168.3.63:8585/static/out/vs/workbench/workbench.web.main.js:1350:9435)
    at v._createInstance (http://192.168.3.63:8585/static/out/vs/workbench/workbench.web.main.js:607:1332)
    at v.createInstance (http://192.168.3.63:8585/static/out/vs/workbench/workbench.web.main.js:607:773)
    at I.createWebviewElement (http://192.168.3.63:8585/static/out/vs/workbench/workbench.web.main.js:1350:21856)
    at B._createInset (http://192.168.3.63:8585/static/out/vs/workbench/workbench.web.main.js:1953:7308)
    at B._initialize (http://192.168.3.63:8585/static/out/vs/workbench/workbench.web.main.js:1953:413)
    at http://192.168.3.63:8585/static/out/vs/workbench/workbench.web.main.js:1949:53
[2022-04-10 11:00:18.073] [window] [error] [Extension Host] (node:2304093) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
```

**Reply by jeanp413** [reply]
@ayan-iiitd you are accessing `http://192.168.3.63:8585`, instead you should use `localhost:8585`, that error is from [here](https://github.com/microsoft/vscode/blob/497a035770bd61298af9cdb74f2a0108bd6e406b/src/vs/workbench/browser/webview.ts#L10-L12)
as it's using `crypto.subtle` API only available in secure contexts (i.e. `https` or `localhost`)

**Reply by ayan2k26** [reply]
@jeanp413 I did try both using `192.168.3.63:3000` and `localhost:3000`. I started with `./openvscode-server --host 192.168.3.63` and in another terminal I did `ssh -L 3000:192.168.3.63:3000 saifk@192.168.3.63`. Now on my local system if I try to login to `http://localhost:3000` it shows forbidden error and in `http://192.168.3.63:3000` it doesn't show ipynb files. If I do not give host, then neither of them connect, nor throw any errors other than `channel 3: open failed: connect failed: Connection refused` in the terminal with the `ssh -L` command.

**Reply by jeanp413** [reply]
@ayan-iiitd are you also using `--without-connection-token` when launching the server? maybe that's why you get the forbidden error if you are not passing the token.
Try:
`./openvscode-server --host 192.168.3.63 --port 3000 --without-connection-token`

**Reply by augustoicaro** [reply]
> @jeanp413 I did try both using `192.168.3.63:3000` and `localhost:3000`. I started with `./openvscode-server --host 192.168.3.63` and in another terminal I did `ssh -L 3000:192.168.3.63:3000 saifk@192.168.3.63`. Now on my local system if I try to login to `http://localhost:3000` it shows forbidden error and in `http://192.168.3.63:3000` it doesn't show ipynb files. If I do not give host, then neither of them connect, nor throw any errors other than `channel 3: open failed: connect failed: Connection refused` in the terminal with the `ssh -L` command. 
> 
As @jeanp413 said looks like a token problem. You can also define your own token with the flag `--connection-token token` and use it in the browser address as `localhost:3000/?tkn=token`. Or use the token provided when you start the server. 

Version 1.65.2 is working as expected and we are using in the production environment at my company. If none of our suggestions works, I recommend you to try the 1.65.2 version.

**Reply by ayan2k26** [reply]
@augustoicaro @jeanp413 both solutions with tokens didn't work. Might have messed up something even. Now when trying to open a ipynb file this is what it shows.

![Screenshot_20220410_220302](https://user-images.githubusercontent.com/59263847/162630096-3506195e-fe7c-4bae-ac33-d9065c213b25.jpeg)


And I have the extensions installed just as before.

![Screenshot_20220410_221117](https://user-images.githubusercontent.com/59263847/162630126-37c46a4b-f42f-468a-a6e9-c41a595da2a9.jpeg)

I have tried uninstalling and reinstalling the extensions as well, but same error persists. Maybe I will give 1.65.2 a try.

**Reply by augustoicaro** [reply]
That's odd, I never get this kind of error before in the versions that I tried. Normally when you don't use the ssh tunnel it will open the file but nothing will appear. You should try the 1.65.2 version and check if the problem persists.

**Reply by ayan2k26** [reply]
@augustoicaro Neither 1.65.2 nor 1.67 insiders worked for me. Some of the things I noticed are that when I tried a different version, the extensions were still installed. the jupyter and python ones. Now the 1.65 version had some language packs installed that needed reloads but it keeps giving the same message even after reloading.

![Screenshot_20220411_110543](https://user-images.githubusercontent.com/59263847/162673327-e2608aca-78d0-4e93-aebe-4eccf158ffb9.jpeg)

And when trying to open any ipynb file I get the Unable to open error msg that I shared a screen of up here and in the `Output/ Log(Remote Extension)` I get this - 

```
[2022-04-11 05:37:46.240] [exthost] [info] Extension host with pid 2319143 started
[2022-04-11 05:37:46.246] [exthost] [info] Lock '/home/saifk/.openvscode-server/data/User/workspaceStorage/449ab27f/vscode.lock': Lock acquired.
[2022-04-11 05:37:46.491] [exthost] [info] ExtensionService#_doActivateExtension vscode.git-base, startup: true, activationEvent: '*', root cause: vscode.github
[2022-04-11 05:37:46.497] [exthost] [info] ExtensionService#_doActivateExtension vscode.debug-auto-launch, startup: true, activationEvent: '*'
[2022-04-11 05:37:46.518] [exthost] [info] ExtensionService#_doActivateExtension vscode.git, startup: true, activationEvent: '*'
[2022-04-11 05:37:46.582] [exthost] [info] ExtensionService#_doActivateExtension vscode.github, startup: true, activationEvent: '*'
[2022-04-11 05:37:46.653] [exthost] [info] ExtensionService#_doActivateExtension vscode.merge-conflict, startup: false, activationEvent: 'onStartupFinished'
[2022-04-11 05:37:46.660] [exthost] [info] Eager extensions activated
[2022-04-11 05:37:51.930] [exthost] [info] ExtensionService#_doActivateExtension vscode.ipynb, startup: false, activationEvent: 'onNotebook:jupyter-notebook}'
[2022-04-11 05:37:51.953] [exthost] [error] Activating extension vscode.ipynb failed due to an error:
[2022-04-11 05:37:51.953] [exthost] [error] Error: Extension 'vscode.ipynb' CANNOT use API proposal: notebookEditor.
Its package.json#enabledApiProposals-property declares: [] but NOT notebookEditor.
 The missing proposal MUST be added and you must start in extension development mode or use the following command line switch: --enable-proposed-api vscode.ipynb
	at E (/home/saifk/openvscode-server-v1.65.2-linux-x64/out/vs/workbench/api/node/extensionHostProcess.js:83:64781)
	at Object.onDidChangeNotebookCells (/home/saifk/openvscode-server-v1.65.2-linux-x64/out/vs/workbench/api/node/extensionHostProcess.js:99:51398)
	at e.ensureAllNewCellsHaveCellIds (/home/saifk/.openvscode-server/extensions/vscode.ipynb-1.62.3/dist/ipynbMain.js:1:141168)
	at t.activate (/home/saifk/.openvscode-server/extensions/vscode.ipynb-1.62.3/dist/ipynbMain.js:1:153425)
	at Function._callActivateOptional (/home/saifk/openvscode-server-v1.65.2-linux-x64/out/vs/workbench/api/node/extensionHostProcess.js:90:16082)
	at Function._callActivate (/home/saifk/openvscode-server-v1.65.2-linux-x64/out/vs/workbench/api/node/extensionHostProcess.js:90:15747)
	at /home/saifk/openvscode-server-v1.65.2-linux-x64/out/vs/workbench/api/node/extensionHostProcess.js:90:13571
	at async Promise.all (index 0)
	at async d.$activate (/home/saifk/openvscode-server-v1.65.2-linux-x64/out/vs/workbench/api/node/extensionHostProcess.js:90:21849)
```

**Reply by augustoicaro** [reply]
🤔
If the 1.65.2 it is working for me, maybe is something related how you are providing the server. How do you provide the server? Docker or local install?

**Reply by jeanp413** [reply]
The `reload required` for built-in extensions was a bug in vscode upstream that got fixed in 1.66 but it's harmless so don't worry about it.
Now the error `Error: Extension 'vscode.ipynb' CANNOT use API proposal: notebookEditor.` is really odd :thinking: , `vscode.ipynb` is a built-in extension so not sure why you would get an error saying `package.json#enabledApiProposals-property` is incorrect
@ayan-iiitd  Try deleting `/home/saifk/.openvscode-server` folder, it will be created automatically when you launch openvscode-server again, it looks it's corrupted somehow as I see the built-in extension `vscode.ipynb-1.62.3` in there `/home/saifk/.openvscode-server/extensions/vscode.ipynb-1.62.3` which is incorrect

**Reply by ayan2k26** [reply]
Deleting `/home/saifk/.openvscode-server` doesn't help. Still getting errors about extension activation failed after installing the python and jupyter extensions.

**Reply by jeanp413** [reply]
Can you share this new errors you get?

**Reply by augustoicaro** [reply]
@ayan-iiitd did you try use the docker version? I saw you your 1st post, and loking your path seems that you are using local install. As everything you tried so far failed, can be a dependency problem, using docker will create a reproducible scenario that will be easier to debug and help you.

Also if it works in docker version just confirms my problem hypothesis.

**Reply by ayan2k26** [reply]
@augustoicaro  Sadly, I cannot use docker. I need access to a lot of files stored locally through code, which I also access outside my code.
And if the docker keeps running, the only way I can get access to that directory is through root which I do not have privileges of.

@jeanp413
> Can you share this new errors you get?

```
[2022-04-11 14:34:09.900] [exthost] [info] Extension host with pid 2333525 started
[2022-04-11 14:34:09.908] [exthost] [info] Lock '/home/saifk/.openvscode-server/data/User/workspaceStorage/-58ad5f89/vscode.lock': Lock acquired.
[2022-04-11 14:34:10.167] [exthost] [info] ExtensionService#_doActivateExtension vscode.git-base, startup: true, activationEvent: '*', root cause: vscode.git
[2022-04-11 14:34:10.173] [exthost] [info] ExtensionService#_doActivateExtension vscode.debug-auto-launch, startup: true, activationEvent: '*'
[2022-04-11 14:34:10.196] [exthost] [info] ExtensionService#_doActivateExtension vscode.git, startup: true, activationEvent: '*'
[2022-04-11 14:34:10.262] [exthost] [info] ExtensionService#_doActivateExtension vscode.github, startup: true, activationEvent: '*'
[2022-04-11 14:34:10.338] [exthost] [info] ExtensionService#_doActivateExtension vscode.merge-conflict, startup: false, activationEvent: 'onStartupFinished'
[2022-04-11 14:34:10.346] [exthost] [info] Eager extensions activated
[2022-04-11 14:34:35.373] [exthost] [info] ExtensionService#_doActivateExtension vscode.ipynb, startup: false, activationEvent: 'onNotebook:jupyter-notebook}'
[2022-04-11 14:34:35.397] [exthost] [error] Activating extension vscode.ipynb failed due to an error:
[2022-04-11 14:34:35.397] [exthost] [error] TypeError: o.notebooks.onDidChangeNotebookCells is not a function
	at e.ensureAllNewCellsHaveCellIds (/home/saifk/.openvscode-server/extensions/vscode.ipynb-1.62.3/dist/ipynbMain.js:1:141168)
	at t.activate (/home/saifk/.openvscode-server/extensions/vscode.ipynb-1.62.3/dist/ipynbMain.js:1:153425)
	at Function._callActivateOptional (/home/saifk/openvscode-server-insiders-v1.67.0-linux-x64/out/vs/workbench/api/node/extensionHostProcess.js:90:16942)
	at Function._callActivate (/home/saifk/openvscode-server-insiders-v1.67.0-linux-x64/out/vs/workbench/api/node/extensionHostProcess.js:90:16606)
	at /home/saifk/openvscode-server-insiders-v1.67.0-linux-x64/out/vs/workbench/api/node/extensionHostProcess.js:90:14430
	at async C._activate (/home/saifk/openvscode-server-insiders-v1.67.0-linux-x64/out/vs/workbench/api/node/extensionHostProcess.js:85:8180)
	at async C._waitForDepsThenActivate (/home/saifk/openvscode-server-insiders-v1.67.0-linux-x64/out/vs/workbench/api/node/extensionHostProcess.js:85:8122)
	at async C._initialize (/home/saifk/openvscode-server-insiders-v1.67.0-linux-x64/out/vs/workbench/api/node/extensionHostProcess.js:85:7486)
[2022-04-11 14:34:52.534] [exthost] [info] ExtensionService#_doActivateExtension ms-python.python, startup: false, activationEvent: 'onLanguage:python'
[2022-04-11 14:34:52.810] [exthost] [info] ExtensionService#_doActivateExtension ms-toolsai.jupyter, startup: false, activationEvent: 'onLanguage:python'
```

**Reply by jeanp413** [reply]
@ayan-iiitd did you delete `/home/saifk/.openvscode-server` folder? I still see `vscode.ipynb-1.62.3` in `extensions` folder, that should never happen, only the extensions you installed manually should be there i.e. jupyter (ms-toolsai.jupyter-x.x.x), jupiter keymap (ms-toolsai.jupyter-keymap-x.x.x), jupyter notebook renderer (ms-toolsai.jupyter-renderers-x.x.x) and python (ms-python.python-x.x.x) (from your screenshot above)
Try deleting it again and check if `vscode.ipynb-1.62.3` is recreated (it should not)

**Reply by ayan2k26** [reply]
@jeanp413 It got recreated when I reinstalled the extensions.

**Reply by jeanp413** [reply]
@ayan-iiitd I just did some tests and it seems a bug in 1.66.1, will investigate, 1.65.2 should work just fine, could try it again, remove `/home/saifk/.openvscode-server` before launching openvscode-server 1.65.2, you should only see the 4 extensions installed in `extensions` folder

**Reply by ayan2k26** [reply]
OK, finally got it to work. I tried a completely new private window, with all extensions disabled. Deleted the existing folder. Reinstalled the Python extension(s). Then I tried opening notebooks, changing kernels and running a line of code, closing the window and starting a new fresh private window then going to the link, then I got the message as if I am opening VSCode for the first time, but all extensions worked, and so did the code. Thanks a lot. I will be looking forward to the updates and also, grateful to be able to find a bug.
![Screenshot_20220413_004025](https://user-images.githubusercontent.com/59263847/163037001-c078e14c-c2c0-4011-a342-4b316b5d55a1.jpeg)
